### DISCUSSION TITLE
Correct Value Showing up in Dev Tools but in the App returns 0

### DISCUSSION THREAD

**Discussion by dafterinc** [Budibase/budibase#9166] [Help]
Hi,
In my app, I am using a javascript function
![image](https://user-images.githubusercontent.com/71251389/209631911-81dacfe5-a7a8-4a98-980f-0afe7c5cb6ba.png)

```js
const pendingPayments = $("Stat Data.Rows").filter(payment => payment["Order Payment Status"] !== "Payment Complete")
var totalPendingPayments = 0
totalPendingPayments = pendingPayments?.reduce( (total, payment) => {
  let pending = payment["Company Freight "] - payment["Company Advance"] || 0
  return total = total + pending
}, 0)	
// Format the total pending payment as INR currency
const formatter = new Intl.NumberFormat('en-IN', {
  style: 'currency',
  currency: 'INR'
})
const formattedTotalPendingPayments = formatter.format(totalPendingPayments)
return formattedTotalPendingPayments
```

This function does not return the value in the app preview.

However when I check the component using DevTools, I can see the correct value in the bindings
![image](https://user-images.githubusercontent.com/71251389/209632080-bd69a3f9-6173-43ba-8264-2db1456cf4f5.png)

What could be the issue that this value is not being shown inside the stat card? It is showing up correctly in the binding therefore the Js functioning correctly, right?

**Comment by melohagan** [comment]
Hey @dafterinc 

It may be an issue with the initial binding value on screen load, i.e. if the rows haven't loaded then it has the value zero. 

I wonder if you added a button that runs that code again, but maybe puts the resultant `formattedTotalPendingPayments` into App State and the Stat Card value was tied to the app state, then would it display the correct value?
This would just be to test that there is an issue with the initial value on load.

Possibly could be related to: https://github.com/Budibase/budibase/discussions/9069
